File: core/src/main/java/com/github/leodan11/customview/core/ReadMoreOption.java
package com.github.leodan11.customview.core;

import android.content.Context;

import androidx.annotation.Nullable;

import com.github.leodan11.customview.core.base.ReadMoreOptionBase;

public class ReadMoreOption extends ReadMoreOptionBase {

    public ReadMoreOption(Context context, int textLength, int textLengthType, String moreLabel, @Nullable ReadOptionListener.More onClickMoreListener, String lessLabel, @Nullable ReadOptionListener.Less onClickLessListener, int moreLabelColor, int lessLabelColor, boolean labelUnderLine, boolean expandAnimation) {
        super(context, textLength, textLengthType, moreLabel, onClickMoreListener, lessLabel, onClickLessListener, moreLabelColor, lessLabelColor, labelUnderLine, expandAnimation);
    }

    public static class Builder extends ReadMoreOptionBase.Builder<ReadMoreOption> {
        public Builder(Context context) {
            super(context);
        }

        @Override
        public ReadMoreOption build() {
            return new ReadMoreOption(
                    context,
                    textLength,
                    textLengthType,
                    moreLabel,
                    onClickMoreListener,
                    lessLabel,
                    onClickLessListener,
                    moreLabelColor,
                    lessLabelColor,
                    labelUnderLine,
                    expandAnimation
            );
        }
    }

    public static final int TYPE_LINE = 1;
    public static final int TYPE_CHARACTER = 2;
}

File: core/src/main/java/com/github/leodan11/customview/core/ReadMoreOption.kt
package com.github.leodan11.customview.core

import android.content.Context
import com.github.leodan11.customview.core.base.ReadMoreOptionBase

class ReadMoreOption(
    context: Context,
    textLength: Int,
    textLengthType: Int,
    moreLabel: String?,
    lessLabel: String?,
    moreLabelColor: Int,
    lessLabelColor: Int,
    labelUnderLine: Boolean,
    expandAnimation: Boolean
) :
    ReadMoreOptionBase(
        context = context,
        textLength = textLength,
        textLengthType = textLengthType,
        moreLabel = moreLabel,
        lessLabel = lessLabel,
        moreLabelColor = moreLabelColor,
        lessLabelColor = lessLabelColor,
        labelUnderLine = labelUnderLine,
        expandAnimation = expandAnimation
    ) {


    /**
     * Creates an [ReadMoreOption] with the arguments supplied to this builder.
     *
     * @property context The parent context
     * @constructor Create empty Builder.
     */
    class Builder(context: Context) : ReadMoreOptionBase.Builder<ReadMoreOption>(context) {
        override fun build(): ReadMoreOption {
            return ReadMoreOption(
                context = context,
                textLength = textLength,
                textLengthType = textLengthType,
                moreLabel = moreLabel,
                lessLabel = lessLabel,
                moreLabelColor = moreLabelColor,
                lessLabelColor = lessLabelColor,
                labelUnderLine = labelUnderLine,
                expandAnimation = expandAnimation
            )
        }
    }

    companion object {
        const val TYPE_LINE: Int = 1
        const val TYPE_CHARACTER: Int = 2
    }

}